### PR TITLE
Change make build to run go build and add install task

### PR DESCRIPTION
**Pull request by endorama**
Follow-up for a conversation @jsoriano, @mtojek and me had about how `make build` works at the moment.

`make build` performs `go install` under the hood. This behaviour is unexpected as `go install` moves the built binary to `GOBIN` folder, which in most system is part of system `PATH`. While this is the purpose of `go install` in the context of `elastic-package` makes a bit troublesome to build binary for development and local testing.

The current workaround is to manually run `go build`, but we loose version information and is undocumented.

To improve the situation this PR moves the current `build` task logic to a separate `install` one and changes the `build` task to perform `go build`, creating the executable in the project folder.

### CODE REVIEW

**Comment by jsoriano** [comment]
Thanks for opening this PR! Could you please also check what documentation needs to be updated?

File: Makefile [hunk 1]
```diff
@@ -5,16 +5,21 @@ VERSION_IMPORT_PATH = github.com/elastic/elastic-package/internal/version
 .PHONY: build
 
 build:
-	go install -ldflags \
+	go build -ldflags \
 	    "-X $(VERSION_IMPORT_PATH).CommitHash=`git describe --always --long --dirty` -X $(VERSION_IMPORT_PATH).BuildTime=`date +%s` -X $(VERSION_IMPORT_PATH).Tag=`(git describe --exact-match --tags 2>/dev/null || echo '') | tr -d '\n'`" \
```
Comment (efd6) [inline Makefile:9]: Can this string be factored out since it appears twice?

Comment (mtojek) [reply Makefile:9]: In general, there is a [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)), but if it isn't a big deal, we can extract this line to the top.

**Review by mtojek — COMMENTED** [review]
You have to adjust also all scripts (.sh) as they refer to the globally installed elastic-package:

```
[2022-01-24T17:32:17.361Z] + make PACKAGE_UNDER_TEST=aws test-check-packages-parallel
[2022-01-24T17:32:17.361Z] PACKAGE_TEST_TYPE=parallel ./scripts/test-check-packages.sh
[2022-01-24T17:32:17.361Z] + trap cleanup EXIT
[2022-01-24T17:32:17.361Z] + OLDPWD=/var/lib/jenkins/workspace/PR-663-2-a78640ec-85fd-454a-a89c-932d6fe3894a/src/github.com/elastic/elastic-package
[2022-01-24T17:32:17.361Z] + for d in test/packages/${PACKAGE_TEST_TYPE:-other}/${PACKAGE_UNDER_TEST:-*}/
[2022-01-24T17:32:17.361Z] + cd test/packages/parallel/aws/
[2022-01-24T17:32:17.361Z] + elastic-package check -v
[2022-01-24T17:32:17.361Z] ./scripts/test-check-packages.sh: line 41: elastic-package: command not found
[2022-01-24T17:32:17.361Z] + cleanup
[2022-01-24T17:32:17.361Z] + r=127
[2022-01-24T17:32:17.361Z] + elastic-package stack dump -v --output build/elastic-stack-dump/check-aws
[2022-01-24T17:32:17.361Z] ./scripts/test-check-packages.sh: line 9: elastic-package: command not found
[2022-01-24T17:32:17.361Z] make: *** [Makefile:79: test-check-packages-parallel] Error 127
```

As a simple solution you can prepend an extra `PATH=$PATH:\`pwd\`/build` (considering that the binary is stored in `build`).

**Comment by endorama** [comment]
@jsoriano I found a reference to `make build` in README, so I updated template and README adding `make install` in https://github.com/elastic/elastic-package/pull/663/commits/96a36915790387487a34dbf2fd50f9addfd581a8

**Comment by endorama** [comment]
@mtojek I preferred to replace `make build` with `make install` as it keeps the same logic tests were using before. But I missed one that I changed in 60ed6bd

File: Makefile (hunk 1 above)
Comment (endorama) [reply Makefile]: I extracted and split the line to give it more clarity in [8182d2f](https://github.com/elastic/elastic-package/pull/663/commits/8182d2f5e5b70b1a1c9b7611187a72a615fdd588)

File: Makefile [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
 
 .PHONY: build
 
 build:
-	go install -ldflags \
-	    "-X $(VERSION_IMPORT_PATH).CommitHash=`git describe --always --long --dirty` -X $(VERSION_IMPORT_PATH).BuildTime=`date +%s` -X $(VERSION_IMPORT_PATH).Tag=`(git describe --exact-match --tags 2>/dev/null || echo '') | tr -d '\n'`" \
-	    github.com/elastic/elastic-package
+	go build \
+			-ldflags "$(VERSION_LDFLAGS)" \
+	    -o elastic-package
```
Comment (mtojek) [inline Makefile:14]: one-liner?

File: Makefile [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
 clean:
 	rm -rf build
 
 format:
 	go run golang.org/x/tools/cmd/goimports -local github.com/elastic/elastic-package/ -w .
 
+install:
+	go install \
+			-ldflags "$(VERSION_LDFLAGS)" \
+	    github.com/elastic/elastic-package
```
Comment (mtojek) [inline Makefile:25]: one-liner?

**Review by mtojek — COMMENTED** [review]
small nit-picks, but I think we're aligned in general :)

File: Makefile (hunk 2 above)
Comment (endorama) [reply Makefile]: done

**Review by mtojek — APPROVED** [review]
LGTM

Feel free to merge if CI is happy.

File: Makefile (hunk 1 above)
Comment (efd6) [reply Makefile]: What you have now is very nice. Thanks.

**Pull request merged**